includes/Modules/Analytics_4.php

```diff
@@ -620,6 +615,26 @@ public static function filter_webdatastream_with_ids( $webdatastream ) {
 		return $obj;
 	}
 
+	/**
+	 * Parses a response, adding the _id and _propertyID params and converting to an array keyed by the propertyID and web datastream IDs.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param GoogleAnalyticsAdminV1alphaListWebDataStreamsResponse[] $response Array of GoogleAnalyticsAdminV1alphaListWebDataStreamsResponse objects.
+	 * @return \stdClass[] Array of models containing _id and _propertyID attributes, keyed by the propertyID.
+	 */
+	public static function filter_webdatastreams_batch( $response ) {
+		$mapped         = array();
+		$results        = array_shift( $response );
+		$webdatastreams = $results->getWebDataStreams();
+		foreach ( $webdatastreams as $webdatastream ) {
+			$value          = self::filter_webdatastream_with_ids( $webdatastream );
+			$key            = $value->_propertyID; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$mapped[ $key ] = array( $value );
```

Comment:
There might be more than one web data stream per property. Create a new GA4 account with 2 or more GA4 properties and many web data streams per property. Then see what this function returns for properties from the new account.

Reply:
https://github.com/google/site-kit-wp/pull/3755/commits/7886d760d2ff3456eaa4e8dd101be76d379d8702